io/zenoh-transport/src/unicast/establishment/ext/auth/mod.rs

```diff
@@ -571,7 +571,10 @@ impl<'a> AcceptFsm for &'a AuthFsm<'a> {
     }
 
     type RecvOpenSynIn = (&'a mut StateAccept, Option<open::ext::Auth>);
-    type RecvOpenSynOut = ();
+
+    #[cfg(feature = "auth_usrpwd")]
```

Comment:
I suggest to have one single type as return type that has multiple fields, one per extension. The inner field than could be conditionally available.